logger.go

```diff
@@ -51,10 +57,22 @@ func (l *Logger) SetWriter(w io.Writer) {
 	l.mu.Unlock()
 }
 
+func (l *Logger) SetFlags(flags int32) {
+	l.mu.Lock()
+	l.flags = l.flags | flags
```

Comment:
The SetFlags is changed to set a flag and another AddFlags method has been added to allow adding flags without taking off the existing ones.